### ISSUE TITLE
[Mac OS X] desktop notification: better use "Alerts" if user interaction is required

### ISSUE THREAD

**Issue by michaelstingl** [owncloud/client#4410]
ownCloud client shows "Banners" to notify users about sync activities.

> **Banners** appear in the upper-right corner of your screen, without interrupting what you're doing. Banners appear  for a short period of time, then slide off screen to the right.
> 
> <img width="300" alt="yosemite_banner" src="https://support.apple.com/library/content/dam/edam/applecare/images/en_US/osx/yosemite_banner.png">

Sometimes, user interaction is required (expiring SAML session). With 2.1.1, those notifications also are displayed as "Banner" and disappear after 2 seconds:

![owncloud-2_1_1](https://cloud.githubusercontent.com/assets/214010/12685079/83ad1864-c6c2-11e5-802d-750df7ec61e6.png)

I would suggest to ~~display~~ trigger so called "Alerts" ~~instead~~ for such notifications.

> **Alerts** are important announcements that appear similar to banners. Alerts remain on your screen until you take action. Many alerts include buttons you can use to interact with them without having to open the app that generated them.
> 
> <img width="300" alt="yosemite_chat_alert" src="https://support.apple.com/library/content/dam/edam/applecare/images/en_US/osx/yosemite_chat_alert.png">

More information about notifications on Mac OS X:
https://support.apple.com/en-us/HT204079

@MTRichards @dragotin What do you think?

00004467

**Comment by MTRichards** [comment]
Interested in @danimo too.

In my experience, there is not universal agreement on this. It tends to be a highly personal and subjective answer based on the users and what they like.

**Comment by michaelstingl** [comment]
@MTRichards Yes, you are right. That's why users can go to "System Preferences" > "Notifications" and configure for each software individually to display alerts as "Banner".

**Comment by MTRichards** [comment]
interesting. Still looking at @danimo  and @dragotin for thoughts, plus @jancborchardt for UI.

**Comment by guruz** [comment]
Which build did you try? AFAIK recent 2.1.1 nightlies should not Show this notification at all but Pop up the auth Dialog
https://github.com/owncloud/client/commit/8486a2fd2be03c7d2eeb0ad8539c23e2bdf5a7b6

@jturcotte please correct me if needed

**Comment by jturcotte** [comment]
In 2.1.1 it should pop together with an shibboleth authentication window, only if the token expired. The notification shouldn't be shown alone on its own like it does in 2.1.0 anymore.

**Comment by MorrisJobke** [comment]
> In 2.1.1 it should pop together with an shibboleth authentication window, only if the token expired. The notification shouldn't be shown alone on its own like it does in 2.1.0 anymore.

This was communicated -> Ticket could be closed.

**Comment by guruz** [comment]
I'm all for putting this in 2.4 (if there is one, else 3.0)

FYI @pmaier1 ?

**Comment by michaelstingl** [comment]
@dragotin did some investigation about advanced Qt libs that could help here. I remember there have been issues with licence incompatbility.

**Comment by guruz** [comment]
@michaelstingl We already do things on OS X natively, don't really need a library for that. https://github.com/owncloud/client/blame/b7809ded23fdba6aa50b6d834b5bd5fc16865eea/src/gui/systray.mm

**Comment by pmaier1** [comment]
> I'm all for putting this in 2.4 (if there is one, else 3.0)
FYI @pmaier1 ?

Agreed that this change is needed, agreed to postpone!

**Comment by guruz** [comment]
Related https://github.com/owncloud/client/issues/4697
